### PR TITLE
HDDS-1442. add spark container to ozonesecure-mr compose files. Contributed by Ajay Kumar.

### PR DESCRIPTION
**Pull request by ajayydv**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 26 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1037 | trunk passed |
| +1 | compile | 68 | trunk passed |
| +1 | mvnsite | 31 | trunk passed |
| +1 | shadedclient | 675 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 22 | dist in the patch failed. |
| +1 | compile | 21 | the patch passed |
| +1 | javac | 21 | the patch passed |
| +1 | mvnsite | 22 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 21 | The patch generated 0 new + 104 unchanged - 132 fixed = 104 total (was 236) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 750 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 22 | dist in the patch passed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 2903 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-746/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/746 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  yamllint  shellcheck  shelldocs  |
| uname | Linux 607d2209fcb4 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 04c0437 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/1/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/1/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/dist U: hadoop-ozone/dist |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml [hunk 1]
```diff
@@ -112,3 +112,13 @@ services:
       HADOOP_CLASSPATH: /opt/ozone/share/ozone/lib/hadoop-ozone-filesystem-lib-current-@project.version@.jar
       WAIT_FOR: rm:8088
     command: ["yarn","timelineserver"]
+  spark:
+    image: ahadoop/spark:v1
```
Comment (xiaoyuyao) [inline hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml]: Can you share the image details of ahadoop/spark:v1? Is it published in docker repo?

Comment (ajayydv) [reply hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml]: Its based on our hadoop docker runner image. You can look at source at https://github.com/ajayydv/docker/tree/spark . We can create a team account for ozone in githib and docker and push this there.

**Comment by ajayydv** [comment]
> Looks good to me but not clear what is the ahadoop/spark:v1 and how is it maintained.
> 
> 1. Where it the source of this image and how can we create PRs? (If you have a simple repo and automated build can be better...)
> 2. Which version of spark is included in the image (v1 tag is not very clear)
> 3. The entrypoint of the dockerimage doesn't allow to start the container easily (docker run ahadoop/spark:v1 bash --> instead of bash the download is started)
Its modified version of our hadoop docker runner. Source is currently in my private repo (https://github.com/ajayydv/docker/tree/spark) but we can push it to public git. Don't think hadoop repo is right place for this as this has mostly spark bits. this uses spark 2.4.3 but it can be overrided via config (SPARK_DOWNLOAD_URL).

File: hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml (hunk 1 above)
Comment (elek) [reply hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml]: It's fine. Just use automated builds: https://docs.docker.com/docker-hub/builds/ 

For automated builds the source will be linked by dockerhub without any effort and easy to check the source.

(Or add a readme with a link to the github project, but automated builds can be better).

Comment (ajayydv) [reply hadoop-ozone/dist/src/main/compose/ozonesecure-mr/docker-compose.yaml]: @elek thanks for documentation link. Added source for automated builds in docker.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 532 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1150 | trunk passed |
| +1 | compile | 77 | trunk passed |
| +1 | mvnsite | 30 | trunk passed |
| +1 | shadedclient | 684 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 20 | dist in the patch failed. |
| +1 | compile | 20 | the patch passed |
| +1 | javac | 20 | the patch passed |
| +1 | mvnsite | 24 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 14 | The patch generated 0 new + 104 unchanged - 132 fixed = 104 total (was 236) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 743 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 17 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 23 | dist in the patch passed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 3518 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-746/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/746 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  yamllint  shellcheck  shelldocs  |
| uname | Linux 59831dd396ef 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a703dae |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/2/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/2/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/dist U: hadoop-ozone/dist |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-746/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by elek** [comment]
Committed. Thanks @ajayydv the contribution.

**Pull request closed without merging**